### PR TITLE
[FEATURE REQUEST] Support "per app" language change on Android 13+

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App: https://github.com/owncloud/android/issues/4082

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: changelog/unreleased/4082
Comment (JuancaG05) [inline changelog/unreleased/4082:1]: This file should be named `4099` 😀

File: owncloudApp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -53,7 +53,8 @@
         android:requestLegacyExternalStorage="true"
         android:resizeableActivity="true"
         android:supportsPictureInPicture="false"
-        android:theme="@style/Theme.ownCloud.Toolbar">
+        android:theme="@style/Theme.ownCloud.Toolbar"
+        android:localeConfig="@xml/locales_config">>
```
Comment (JuancaG05) [inline owncloudApp/src/main/AndroidManifest.xml:57]:
```suggestion
        android:localeConfig="@xml/locales_config">
```
Or am I missing something? 🤔

File: owncloudApp/src/main/res/xml/locales_config.xml
Comment (JuancaG05) [inline owncloudApp/src/main/res/xml/locales_config.xml:1]: Not sure if some variants are included here when their identifier doesn't appear in this list 🤔. For example:

Captura de pantalla 2023-07-19 a las 10 34 49

Not the most populated languages of the app but just not to lose them in this feature. Maybe if you tried and it doesn't work it's not allowed to add them... but I think it's worth it to take a look with a quick research over the Internet or some other quick tests 🚀

File: owncloudApp/src/main/res/xml/locales_config.xml [hunk 2]
```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0" encoding="utf-8"?>
+<locale-config xmlns:android="http://schemas.android.com/apk/res/android">
+    <locale android:name="en-US"/> <!-- English (United States) -->
+    <locale android:name="en-GB"/> <!-- English (United Kingdom) -->
+    <locale android:name="es"/> <!-- Spanish (Spain) -->
+    <locale android:name="ca"/> <!-- Catalan (Catalonia) -->
+    <locale android:name="fr"/> <!-- French (France) -->
```
Comment (JuancaG05) [inline owncloudApp/src/main/res/xml/locales_config.xml:7]: This kind of comments are so useful, I like them a lot 💯

File: owncloudApp/src/main/res/xml/locales_config.xml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    <locale android:name="uk"/> <!-- Ukrainian (Ukraine) -->
+    <locale android:name="ur"/> <!-- Urdu (Pakistan) -->
+    <locale android:name="uz"/> <!-- Uzbek (Uzbekistan) -->
+    <locale android:name="vi"/> <!-- Vietnamese (Vietnam) -->
+    <locale android:name="wo"/> <!-- Wolof (Senegal) -->
+    <locale android:name="xh"/> <!-- Xhosa (South Africa) -->
+    <locale android:name="yo"/> <!-- Yoruba (Nigeria) -->
+    <locale android:name="zh"/> <!-- Chinese (China) -->
+    <locale android:name="zu"/> <!-- Zulu (South Africa) -->
+</locale-config>
```
Comment (JuancaG05) [inline owncloudApp/src/main/res/xml/locales_config.xml:155]: Let's finish with a blank line 👍

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Cool! This will be very useful. I left some suggestions and comments here @Aitorbp 🤠

File: owncloudApp/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Aitorbp) [reply owncloudApp/src/main/AndroidManifest.xml:57]: Done

File: owncloudApp/src/main/res/xml/locales_config.xml [hunk 4]
```diff
@@ -0,0 +1,147 @@
+<?xml version="1.0" encoding="utf-8"?>
+<locale-config xmlns:android="http://schemas.android.com/apk/res/android">
+    <locale android:name="af"/> <!-- Afrikaans (South Africa) -->
+    <locale android:name="am"/> <!-- Amharic (Ethiopia) -->
+    <locale android:name="ar"/> <!-- Arabic (Arab World) -->
+    <locale android:name="arn"/> <!-- Mapudungun (Chile) -->
+    <locale android:name="as"/> <!-- Assamese (India) -->
```
Comment (JuancaG05) [inline owncloudApp/src/main/res/xml/locales_config.xml:7]: We should place here first at least the 3 ultimate fallback languages, these are: `en`, `de` and `es`

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Last changes here and we'll be ready to go! @Aitorbp

File: owncloudApp/src/main/res/xml/locales_config.xml
Comment (JuancaG05) [reply owncloudApp/src/main/res/xml/locales_config.xml:1]: Test comment

**Review by JuancaG05 — APPROVED** [review]
Nice job @Aitorbp! 💯

**Comment by jesmrec** [comment]
A couple of checks:

- [X] Android 13: Check that `ownCloud` is an available option in  device Settings > System > Languages & input > App Languages
- [X] Android 13: Check that app could be set up in a different languages by using the new option available in Settings > System > Languages & input > App Languages
- [X] Android 13: Check that a missing translation in the chosen language is fallen back to the default system language
- [X] Android < 13: Legacy behaviour: app getting device set up language.

Approved on my side and ready to go!!

File: owncloudApp/src/main/res/xml/locales_config.xml
Comment (JuancaG05) [reply owncloudApp/src/main/res/xml/locales_config.xml:1]: When `bn` is included among the available languages it's not necessary to add its variants, the device will show them

**Pull request merged**